packages/fxa-auth-db-mysql/lib/db/mysql.js

```diff
@@ -1192,34 +1192,29 @@ module.exports = function(log, error) {
         // Enforce sane defaults on every new connection.
         // These *should* be set by the database by default, but it's nice
         // to have an additional layer of protection here.
-        connection.query('SELECT @@sql_mode AS mode;', (err, rows) => {
-          if (err) {
-            return reject(err);
-          }
-
-          const currentModes = rows[0]['mode'].split(',');
-          this.requiredModes.forEach(requiredMode => {
-            if (currentModes.indexOf(requiredMode) === -1) {
-              currentModes.push(requiredMode);
-            }
-          });
-
-          const newMode = currentModes.join(',');
-          connection.query(`SET SESSION sql_mode = '${newMode}';`, err => {
-            if (err) {
-              return reject(err);
-            }
-
-            connection.query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;', err => {
-              if (err) {
-                return reject(err);
+        const query = P.promisify(connection.query, { context: connection });
+        return resolve(
+          (async () => {
+            // Always communicate timestamps in UTC.
+            await query("SET time_zone = '+00:00'");
+            // Always use full 4-byte UTF-8 for communicating unicode.
+            await query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;');
+            // Always have certain modes active. The complexity here is to
+            // preserve any extra modes active by default on the server.
+            const rows = await query('SELECT @@sql_mode AS mode;');
+            const currentModes = rows[0]['mode'].split(',');
+            for (const requiredMode of this.requiredModes) {
+              if (currentModes.indexOf(requiredMode) === -1) {
+                currentModes.push(requiredMode);
               }
-
-              connection._fxa_initialized = true;
-              resolve(connection);
-            });
-          });
-        });
+            }
```

Comment:
IIUC, I suppose this could be written as `currentModes = [...new Set([...currentModes, ...this.requiredModes])]`

Reply:
Thanks @chenba!

I believe the current approach deliberately tries to preserve the order of the mode flags although I don't know why and it's probably just from not quite trusting MySQL not to have some nonsense ordering-related problem. I'll add a comment to this effect in the source.